src/m365/commands/request.ts

```diff
@@ -0,0 +1,187 @@
+import { Logger } from '../../cli';
+import Command from '../../Command';
+import GlobalOptions from '../../GlobalOptions';
+import request from '../../request';
+import commands from './commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  url: string;
+  method?: string;
+  resource?: string;
+  body?: string;
+  filePath?: string;
+}
+
+class RequestCommand extends Command {
+  private allowedMethods: string[] = ['get', 'post', 'put', 'patch', 'delete', 'head', 'options'];
+
+  public get name(): string {
+    return commands.REQUEST;
+  }
+
+  public get description(): string {
+    return 'Executes the specified web request using CLI for Microsoft 365';
+  }
+
+  public allowUnknownOptions(): boolean | undefined {
+    return true;
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      const properties: any = {
+        method: args.options.method || 'get',
+        resource: typeof args.options.resource !== 'undefined',
+        accept: args.options.accept || 'application/json',
+        body: typeof args.options.body !== 'undefined',
+        filePath: typeof args.options.filePath !== 'undefined'
+      };
+
+      const unknownOptions: any = this.getUnknownOptions(args.options);
+      const unknownOptionsNames: string[] = Object.getOwnPropertyNames(unknownOptions);
+      unknownOptionsNames.forEach(o => {
+        properties[o] = typeof unknownOptions[o] !== 'undefined';
+      });
+
+      Object.assign(this.telemetryProperties, properties);
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --url <url>'
+      },
+      {
+        option: '-m, --method [method]',
+        autocomplete: this.allowedMethods
+      },
+      {
+        option: '-r, --resource [resource]'
+      },
+      {
+        option: '-b, --body [body]'      
+      },
+      {
+        option: '-p, --filePath [filePath]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const currentMethod = args.options.method || 'get';
+        if (this.allowedMethods.indexOf(currentMethod) === -1) {
+          return `${currentMethod} is not a valid value for method. Allowed values: ${this.allowedMethods.join(', ')}`;
+        }
+
+        if (args.options.body && (!args.options.method || args.options.method === 'get')) {
+          return 'Specify a different method when using body';
+        }
+
+        if (args.options.body && !args.options['content-type']) {
+          return 'Specify the content-type when using body';
+        }
+
+        if (args.options.filePath && !fs.existsSync(path.dirname(args.options.filePath))) {
+          return 'Specified filePath to save the file does not exist';
```

Comment:
Let's reword it to make it clear that the folder specified in the file path doesn't exist